numba/tests/test_gdb.py

```diff
@@ -0,0 +1,136 @@
+"""
+Tests gdb bindings
+"""
+from __future__ import print_function
+import ctypes
+import os
+import subprocess
+import sys
+import threading
+
+from numba import njit, gdb, gdb_init, gdb_breakpoint, prange, testing, config
+from numba import unittest_support as unittest
+
+from .support import (TestCase, override_env_config, captured_stdout, tag)
+from .test_parfors import skip_unsupported as parfors_skip_unsupported
+from .test_parfors import linux_only
+
+_linux = sys.platform.startswith('linux')
+
+_gdb_cond = os.environ.get('GDB_TEST', None) == '1'
+needs_gdb_harness = unittest.skipUnless(_gdb_cond, "needs gdb harness")
+_gdbloc = config.GDB_BINARY
+_has_gdb = (os.path.exists(_gdbloc) and os.path.isfile(_gdbloc))
+needs_gdb = unittest.skipUnless(_has_gdb, "gdb binary is required")
+
+
+@linux_only
+class TestGdbBindImpls(TestCase):
```

Comment:
I don't understand what these tests are doing.  Can you add some comments?

Reply:
Done in https://github.com/numba/numba/commit/84cf91cdd771f39bb18db2f54d16e8aa5db97206#diff-44a34807380623154377a065af594213R31